File: backend/internal/entity/quiz.go
package entity

type Quiz struct {
    ID         string          `bson:"_id" json:"id"`
    Name       string          `bson:"name"`
    Questions  []QuizQuestion `bson:"questions"`
}

type QuizQuestion struct {
    ID      string   `bson:"id" json:"id"`
    Name    string   `bson:"name" json:"name"`
    Choices []Choice `bson:"choices" json:"choices"`
}

type Choice struct {
    ID      string `bson:"id" json:"id"`
    Name    string `bson:"name" json:"name"`
    Correct bool   `bson:"correct" json:"correct"`
}

File: frontend/src/model/quiz.ts
export interface QuizChoice {
  id: string;
  name: string;
  correct: boolean;
}

export interface Pplayer{
	id:string;
	name:string;
}

export interface QuizQuestion {
  id: string;
  name: string;
  choices: QuizChoice[];
}

export interface Quiz {
  id: string;
  Name: string;
  questions: QuizQuestion[];
}
